### PR TITLE
Reduce zoom factor when zooming with mouse wheel

### PR DESCRIPTION
**Pull request by LWinterberg**
Part 1 of #4212

### CODE REVIEW

File: src/ProjectWindow.cpp [hunk 1]
```diff
@@ -481,7 +481,7 @@ mutable -> unsigned {
 #endif
 
       wxCoord xTrackEnd = viewInfo.TimeToPosition( audioEndTime );
-      viewInfo.ZoomBy(pow(2.0, steps));
+      viewInfo.ZoomBy(pow(pow(2.0, 1/4.0), steps));
```
Comment (Paul-Licameli) [inline src/ProjectWindow.cpp:484]: Or simply `pow(2.0, steps/4.0)`

**Review by Paul-Licameli — APPROVED** [review]

**Review by Paul-Licameli — APPROVED** [review]
Even better now.

**Comment by LWinterberg** [comment]
I'd like a test from the @Tantacrul to see if this feels better to him

**Comment by SteveDaulton** [comment]
This feels worse for me as it  makes zooming much slower. For example, to zoom from a 30 second view to sample level view requires three mouse scrolls with previous version, and 10 mouse scrolls (tedious) with this patch.

Personally I have no problem with the mouse zoom in previous versions, and due to the absence of feedback to the contrary, I suspect that the vast majority of users agree.

**If** mouse zoom is too coarse for some users (I guess that is the case for [LWinterberg](https://github.com/LWinterberg)), then it may be worth making this a user preference. Personally I'd opt for the same zoom scrolling as Audacity has had for the last decade.

**Comment by Paul-Licameli** [comment]
> This feels worse for me as it makes zooming much slower. For example, to zoom from a 30 second view to sample level view requires three mouse scrolls with previous version, and 10 mouse scrolls (tedious) with this patch.
> 
> Personally I have no problem with the mouse zoom in previous versions, and due to the absence of feedback to the contrary, I suspect that the vast majority of users agree.
> 
> **If** mouse zoom is too coarse for some users (I guess that is the case for [LWinterberg](https://github.com/LWinterberg)), then it may be worth making this a user preference. Personally I'd opt for the same zoom scrolling as Audacity has had for the last decade.

A user preference is always a possibility.

Just lately I was using CTRL plus mouse wheel (or actually, pinch and spread on my touch pad instead of a wheel), while the pointer was in the vertical ruler of the spectrogram view.  Maybe little known, but you can pan (with Shift) and zoom vertical scales too.  And I the defaults were too coarse for my liking.

**Comment by Paul-Licameli** [comment]
I wonder if we could even experiment with some kind of accelerating zoom?  Small steps (logarithmically that is) at first, but larger if the next mouse click arrives in less than some elapsed time?  Complicating the implementation of course.

**Comment by Paul-Licameli** [comment]
Anyway if the multiplier should be some constant but not 2, I suggested it should be a root of 2 so you get the exact twofold multiplier with a few wheel clicks.  It’s now the fourth root, but maybe the cube root is a compromise.

**Comment by LWinterberg** [comment]
https://github.com/audacity/audacity/actions/runs/4027212455 - 4th root (4 scroll ticks = 1 current scroll tick)
https://github.com/audacity/audacity/actions/runs/4038643540 - 3rd root (3 scroll ticks = 1 current scroll tick)

The current behavior is definitely too sensitive. It takes a single scroll to go from default zoom to half a day. With a free-spinning mousewheel, I can zoom from one extreme to the other faster than I can put my finger down to stop the scroll midway. With a trackpad, taking off the fingers from the pad can cause a tick (rarely two) of zooming, which throws off the zoom level by up to an order of magnitude. 

With the 3rd root, a single scroll goes from default zoom to about a song length. With a free-spinning wheel I now have hope of catching it at a reasonable level, and the trackpad errors are nowhere near as significant.

**Comment by Tantacrul** [comment]
@LWinterberg - they're both quite good. I marginally prefer: https://github.com/audacity/audacity/actions/runs/4027212455

As for hiding this change behind a setting. I think that will only unfortunately do two undesirable things

- It won't solve the quite poor performance of the mouse wheel when zooming (responsiveness is part 1 of the issue)
- New settings just introduce more complexity that 95% of our audience will never discover

You can't make changes of any kind without _someone_ being unhappy with it.

**Comment by SteveDaulton** [comment]
> As for hiding this change behind a setting.

Adding a preference does not in any way "hide" the change if you make your preferred setting the default. It just acknowledges that some users may prefer mouse zoom the way that it has been for the last decade.

> You can't make changes of any kind without _someone_ being unhappy with it.

Surely, when some users prefer a feature one way, and other users prefer it another, then it makes sense for it to be a user preference.

**Comment by Tantacrul** [comment]
I think the simplest summary I can give is:

- I doubt too many people will object (or even notice) a slightly less responsive setting for zoom.
- If I'm wrong, we can create a setting for it
- Regardless, the revised version should be the default (unless my finger is completely off the pulse and most users totally hate it)

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @LWinterberg 's latest branch:  audacity-win-3.3.0-alpha-20230129+20dfd68-x64

I find myself liking the less sensitive mouse-wheel scrolling.

I do note that the old zoom factor of 2x still applies to the zoom in/out buttons and commands - and I like that this zoom factor is retained for the menu commands and Edit toolbar buttons.

**Comment by petersampsonaudacity** [comment]
> This feels worse for me as it makes zooming much slower. For example, to zoom from a 30 second view to sample level view requires three mouse scrolls with previous version, and 10 mouse scrolls (tedious) with this patch.

This only takes me (with Leo's branch build) 5 mouse scrolls to go from 30secs to sample level view at full screen size - and only 4 mouse scrolls at default window size,  where a mouse scroll is a full push of the wheel i.e. mutiple mouse-wheel clicks and not a single click.

I also note that if I spin the mouse-wheel forward fast with Ctrl held down the it zooms straight to maximum zoom individual sample level.  Is this an easter-egg?

**Comment by petersampsonaudacity** [comment]
I've just run this past my "Mystery-Shopper" - she too thinks the less aggressive mouse zoom is much better ...

**Comment by Tantacrul** [comment]
We're getting dangerously close to unanimous agreement here. I feel weird.

**Comment by Ryan-Miller-Audacity** [comment]
Tested on Windows and MacOS and confirmed that Zoom factor when zooming with mouse wheel has been reduced and features a much higher degree of usability for general users.

**Pull request merged**